### DISCUSSION TITLE
is there a way to customize the keymap by using self defined js function?

### DISCUSSION THREAD

**Discussion by OriLiMu** [gdh1995/vimium-c#1158] [Q&A] [answered]
I know in customize search engine can make this. But I want to make it quicker by just using keymap like <kbd>ym</kbd>. Any way to make it?

**Comment by gdh1995** [comment, marked as answer]
sorry I can not get what you want.

If you want to run a `javascript:` URL using a key mapping rule, then you may use:
- `map xxx openUrl url="javascript:"`
- or `map xxx openBookmark title="the-exact-title-of-a-target-bookmark"`

**Reply by liby** [reply]
I guess he might want something like [Surfingkeys' feature](https://github.com/brookhong/Surfingkeys#:~:text=Surfingkeys%20is%20created%20with%20all%20settings%20described%20in%20Javascript).

```
api.mapkey('<Ctrl-y>', 'Show me the money', function() {
    api.Front.showPopup('a well-known phrase uttered by characters in the 1996 film Jerry Maguire (Escape to close).');
});
```

**Reply by gdh1995** [reply]
No, because this usage is not encouraged in Vimium C. So please use `openUrl url="<JS-url>"` to align with other key mappings.
